starknet_py/contract.py

```diff
@@ -1012,9 +1012,15 @@ async def deploy_contract_v3(
             auto_estimate=auto_estimate,
         )
 
-        deployed_contract = Contract(
-            provider=account, address=address, abi=abi, cairo_version=cairo_version
-        )
+        if abi:
```

Comment:
I think we should use `if abi is not None`, because if abi is an empty list, then `if abi` condition will not be met.
```suggestion
        if abi is not None:
```